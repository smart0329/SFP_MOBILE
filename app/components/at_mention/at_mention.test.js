// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';
import AtMention from './at_mention.js';

describe('AtMention', () => {
    const baseProps = {
        usersByUsername: {},
        mentionKeys: [{key: 'John.Smith'}, {key: 'Jane.Doe'}],
        teammateNameDisplay: '',
        mentionName: 'John.Smith',
        mentionStyle: {color: '#ff0000'},
        theme: {},
    };

    test('should match snapshot, no highlight', () => {
        const wrapper = shallow(
            <AtMention {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot, with highlight', () => {
        const wrapper = shallow(
            <AtMention {...baseProps}/>,
        );

        wrapper.setState({user: {username: 'John.Smith'}});
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot, without highlight', () => {
        const wrapper = shallow(
            <AtMention {...baseProps}/>,
        );

        wrapper.setState({user: {username: 'Victor.Welch'}});
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});