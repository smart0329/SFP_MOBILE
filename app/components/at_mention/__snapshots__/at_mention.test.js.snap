// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtMention should match snapshot, no highlight 1`] = `
<Text>
  @John.Smith
</Text>
`;

exports[`AtMention should match snapshot, with highlight 1`] = `
<Text
  onLongPress={[Function]}
  onPress={[Function]}
>
  <Text
    style={null}
  >
    @John.Smith
  </Text>
  
</Text>
`;

exports[`AtMention should match snapshot, without highlight 1`] = `
<Text
  onLongPress={[Function]}
  onPress={[Function]}
>
  <Text
    style={
      Object {
        "color": "#ff0000",
      }
    }
  >
    @Victor.Welch
  </Text>
  
</Text>
`;
